deepchem/data/data_loader.py

```diff
@@ -875,9 +875,88 @@ class provides convenience files to lead FASTA data and
   learning tasks.
   """
 
-  def __init__(self):
-    """Initialize loader."""
-    pass
+  def __init__(self,
+               featurizer=OneHotFeaturizer,
+               charset: Union[str, tuple, None] = "ATCGN",
+               max_length: Optional[int] = None,
+               auto_add_annotations: bool = False):
+    """Initialize FASTALoader.
+
+    Parameters
+    ----------
+    featurizer: Featurizer (default: OneHotFeaturizer)
+      The Featurizer to be used for the loaded FASTA data.
+
+    protein: bool (default: False)
+      Whether or not the sequence passed in is a protein sequence. If False,
+      it is treated as a nucleic acid sequence.
+
+    charset: Union[str, tuple, None] (default: "ATCGN")
+      The charset used in the loaded FASTA file.
+      str arguments "protein", "nucleic", and "ATCGN" will respectively load
+      a FASTA format protein sequence character set, FASTA format nucleic acid
+      sequence character set, and ATCG character set.
+
+      If you choose to pass in a tuple, the tuple will be directly used as the
+      charset during featurization.
+
+      Pass None to charset not pass a charset to the featurizer.
+
+    max_length: Optional[int] (default: None)
+      max_length should be equal to or larger than the length of the longest
+      string that is being featurized. max_length is passed to the featurizer.
+      OneHotFeaturizer pads all strings to max_length with spaces.
+
+      Pass -1 to max_length for the loader to not pass a max_length to the featurizer.
+
+    auto_add_annotations: bool (default False)
+      Whether create_dataset will automatically add [CLS] and [SEP] annotations
+      to the sequences it reads in order to assist tokenization.
+      Keep False if your FASTA file already includes [CLS] and [SEP] annotations.
+   """
+    charsets = {
+        "protein":
+        ('A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
+         'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '*', '-'),
+        "nucleic": ('A', 'C', 'G', 'T', 'U', '(i)', 'R', 'Y', 'K', 'M', 'S',
+                    'W', 'B', 'D', 'H', 'V', 'N', '-'),
+        "ATCGN": ('A', 'T', 'C', 'G'),
+    }
+
+    # Initialize instance variables
+    self.max_length = max_length
+    self.auto_add_annotations = auto_add_annotations
+    if isinstance(charset, str):
+      try:
+        self.charset = charsets[charset]
+      except KeyError:
+        logger.warning("Charset is invalid string. Treating as None...")
+        self.charset = None
+    elif isinstance(charset, tuple):
+      self.charset = charset
+    else:
+      self.charset = None
+
+    # Set user defined featurizer
+    self.user_specified_features = None
+    if isinstance(featurizer, UserDefinedFeaturizer):
+      self.user_specified_features = featurizer.feature_fields
+
+    # Initialize featurizer
+    try:
+      if self.charset is not None and self.max_length != -1:
+        featurizer = featurizer(
```

Comment:
You should have a check `if featurizer == None` and explicitly just use `OneHotFeaturizer` here

Reply:
![image](https://user-images.githubusercontent.com/54920181/123180121-9d70e700-d4c5-11eb-9b2e-4ac9d3b36e76.png)
It now looks like this.

Reply:
Maybe this design is shortsighted and I should just have the user pass in something like OneHotFeaturizer(charset, max_length) instead?